lib/irb/completion.rb

```diff
@@ -453,16 +453,30 @@ def retrieve_completion_data(input, bind:, doc_namespace:)
         else
           candidates = (bind.eval_methods | bind.eval_private_methods | bind.local_variables | bind.eval_instance_variables | bind.eval_class_constants).collect{|m| m.to_s}
           candidates |= RubyLex::RESERVED_WORDS.map(&:to_s)
-          candidates.grep(/^#{Regexp.quote(input)}/).sort
+          safe_grep(candidates, /^#{Regexp.quote(input)}/).sort
         end
       end
     end
 
     # Set of available operators in Ruby
     Operators = %w[% & * ** + - / < << <= <=> == === =~ > >= >> [] []= ^ ! != !~]
 
+    def safe_grep(candidates, pattern)
+      target_encoding = Encoding.default_external
+      candidates.filter_map do |candidate|
+        next unless candidate
+
+        converted = candidate.encoding == target_encoding ? candidate : candidate.encode(target_encoding)
```

Comment:
Encoding conversion is more than safe-grep. I think this method should only do safe grep (`match?` and `rescue`) for simplicity.

Encoding conversion is already done in line 205.
```ruby
def completion_candidates(preposing, target, postposing, bind:)
  ...
  completion_data = retrieve_completion_data(target, bind: bind, doc_namespace: false).compact.filter_map do |i|
    i.encode(Encoding.default_external)
  rescue Encoding::UndefinedConversionError
    # If the string cannot be converted, we just ignore it
    nil
  end
  ...
end
```